apollo-federation/src/connectors/json_selection/location.rs

```diff
@@ -19,10 +20,42 @@ use super::ParseResult;
 // parsing and then only set Some(message) when we need to report an error, so
 // we would not be cloning long String messages very often (and the rest of the
 // Span fields are cheap to clone).
-pub(crate) type Span<'a> = LocatedSpan<&'a str, Option<&'static str>>;
+pub(crate) type Span<'a> = LocatedSpan<&'a str, SpanExtra>;
+
+#[derive(Debug, PartialEq, Eq, Clone, Hash)]
+pub(crate) struct SpanExtra {
+    pub(super) spec: ConnectSpec,
+    pub(super) errors: Vec<String>,
+}
```

Comment:
Here's why we need more `.clone()`s of input `Span` values used by the parser: we went from passing around a `LocatedSpan<&'a str, Option<&'static str>>` type that was automatically `Copy`, because it was just tracking `&str` references pointing into the original source (or to static error message strings), to passing around a `LocatedSpan<&'a str, SpanExtra>`, which is still just as cheap to clone with regard to the `&'a str` part, but now requires `Clone` instead of `Copy` because `SpanExtra` has owned data.

Since `ConnectSpec` is `Copy`, I suppose we might be able to go back to something like a singular `error: &'static str` so the whole `SpanExtra` struct could be `Copy` again.